### PR TITLE
[JSON-API] User management endpoint adjustments

### PR DESCRIPTION
**Pull request by realvictorprm**
Adjusts the return of the createUser & deleteUser endpoints to be an empty object (instead of true). Furthermore the rights field for creating a user was made optional such that a payload `{ "userId": "nice.user" }` is enough to create a user.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/domain.scala [hunk 1]
```diff
@@ -174,7 +174,7 @@ object domain extends com.daml.fetchcontracts.domain.Aliases {
   final case class CreateUserRequest(
       userId: String,
       primaryParty: Option[String],
-      rights: List[UserRight],
+      rights: Option[List[UserRight]],
```
Comment (stefanobaghino-da) [inline ledger-service/http-json/src/main/scala/com/digitalasset/http/domain.scala:177]: What is the difference in semantics between a `None` and a `Some(Nil)`?

**Review by stefanobaghino-da — COMMENTED** [review]
Looks good, one doubt, leaving it for @S11001001 for approval (unless this is urgent).

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/domain.scala (hunk 1 above)
Comment (realvictorprm) [reply ledger-service/http-json/src/main/scala/com/digitalasset/http/domain.scala:177]: There isn't any, however to not have to write json encoders/decoders by hand I decided to adjust the type to this.

**Comment by realvictorprm** [comment]
> Looks good, one doubt, leaving it for @S11001001 for approval (unless this is urgent).

I would like to merge this today so I can merge the docs pr too today and have this in best case in within January :smile:

**Review by S11001001 — APPROVED** [review]

**Pull request merged**